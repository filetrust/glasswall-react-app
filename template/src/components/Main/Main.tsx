import React from "react";

import styles from "./Main.module.scss";

export interface MainProps {
    expanded: boolean,
    showTitle: boolean,
    title?: string,
    children: React.ReactNode
}

const Main = (props: MainProps) => {

    if (props.showTitle && (props.title === undefined || props.title === null)) {
        console.error("showTitle is set to true, but no title was supplied to <Main>.");
    }

    return (
        <>
            {props.showTitle &&
                <h1 className={`${styles.pageHeading} ${props.expanded ? styles.expanded : ""}`}>
                    {props.title}
                </h1>
            }

            <div className={`${styles.main} ${props.expanded ? styles.expanded : ""} ${props.showTitle ? styles.showTitle : ""}`}>

                <div className={styles.content}>
                    {props.children}
                </div>
            </div>
        </>
    );
};

export default Main;