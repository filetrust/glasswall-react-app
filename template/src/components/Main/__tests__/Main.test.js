import React from "react";
import { mount } from "enzyme";
import TestRenderer from "react-test-renderer";
import Main from "../Main";

const paragraph = <p key="paragraph">Test paragraph.</p>;

const button = <button key="button">Test Button</button>;

const props = {
    expanded: true,
    showTitle: true,
    title: "Test Title",
    children: [
        paragraph,
        button
    ]
};

test("Main_Snapshot", () => {
    // Arrange
    const component = TestRenderer.create(<Main {...props} />);

    // Act
    let tree = component.toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
});

test("Displays_Correct_Props", () => {
    // Arrange
    // Act
    const mainComponent = mount(<Main {...props} />);

    // Assert
    expect(mainComponent.prop("expanded")).toEqual(props.expanded);
    expect(mainComponent.prop("showTitle")).toEqual(props.showTitle);
    expect(mainComponent.prop("title")).toEqual(props.title);
    expect(mainComponent.prop("children")[0]).toEqual(paragraph);
    expect(mainComponent.prop("children")[1]).toEqual(button);
});