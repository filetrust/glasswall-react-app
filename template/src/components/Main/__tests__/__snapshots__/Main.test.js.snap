// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main_Snapshot 1`] = `
Array [
  <h1
    className="pageHeading expanded"
  >
    Test Title
  </h1>,
  <div
    className="main expanded showTitle"
  >
    <div
      className="content"
    >
      <p>
        Test paragraph.
      </p>
      <button>
        Test Button
      </button>
    </div>
  </div>,
]
`;
